java/ql/test/library-tests/frameworks/spring/Test.java

```diff
@@ -0,0 +1,125 @@
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.Locale;
+import java.lang.String;
+
+class Test {
+  String taint() { return "tainted"; }
+
+  String[] taintArray() { return null; }
+
+  Locale taintLocale() { return null; }
+
+  Collection<String> taintedCollection() { return null; }
+
+  Enumeration<String> taintedEnumeration() { return null; }
+
+  void sink(Object o) {}
+
+  void test() throws Exception {
+    sink(StringUtils.addStringToArray(null, taint())); // $hasTaintFlow
+
+    sink(StringUtils.addStringToArray(taintArray(), "")); // $hasTaintFlow
+
+    sink(StringUtils.applyRelativePath("/", taint())); // $hasTaintFlow
+    sink(StringUtils.applyRelativePath(taint(), "../../test")); // $hasTaintFlow
+
+    sink(StringUtils.arrayToCommaDelimitedString(taintArray())); // $hasTaintFlow
+
+    sink(StringUtils.arrayToDelimitedString(taintArray(), ":")); // $hasTaintFlow
+    sink(StringUtils.arrayToDelimitedString(null, taint())); // $hasTaintFlow
+
+    sink(StringUtils.capitalize(taint())); // $hasTaintFlow
+
+    sink(StringUtils.cleanPath(taint())); // $hasTaintFlow
+
+    sink(StringUtils.collectionToCommaDelimitedString(taintedCollection()));  // $hasTaintFlow
+
+    sink(StringUtils.collectionToDelimitedString(taintedCollection(), ":")); // $hasTaintFlow
+    sink(StringUtils.collectionToDelimitedString(null, taint())); // $hasTaintFlow
+
+    sink(StringUtils.collectionToDelimitedString(taintedCollection(), ":", "", "")); // $hasTaintFlow
+    sink(StringUtils.collectionToDelimitedString(null, taint(), "", "")); // $hasTaintFlow
+    sink(StringUtils.collectionToDelimitedString(null, ":", taint(), "")); // $hasTaintFlow
+    sink(StringUtils.collectionToDelimitedString(null, ":", "", taint())); // $hasTaintFlow
+
+    sink(StringUtils.commaDelimitedListToSet(taint())); // $hasTaintFlow
+
+    sink(StringUtils.commaDelimitedListToStringArray(taint())); // $hasTaintFlow
+
+    sink(StringUtils.concatenateStringArrays(taintArray(), null)); // $hasTaintFlow
+    sink(StringUtils.concatenateStringArrays(null, taintArray())); // $hasTaintFlow
+
+    sink(StringUtils.delete(taint(), "")); // $hasTaintFlow
+
+    sink(StringUtils.deleteAny(taint(), "")); // $hasTaintFlow
+
+    sink(StringUtils.delimitedListToStringArray(taint(), ":")); // $hasTaintFlow
+    sink(StringUtils.delimitedListToStringArray(taint(), ":", ".")); // $hasTaintFlow
+
+    sink(StringUtils.getFilename(taint())); // $hasTaintFlow
+
+    sink(StringUtils.getFilenameExtension(taint())); // $hasTaintFlow
+
+    sink(StringUtils.mergeStringArrays(taintArray(), null)); // $hasTaintFlow
+    sink(StringUtils.mergeStringArrays(null, taintArray())); // $hasTaintFlow
+
+    sink(StringUtils.parseLocale(taint()));
+
+    sink(StringUtils.parseLocaleString(taint()));
+
+    // sink(StringUtils.parseTimezoneString(taint())); // does not exist in spring 5.2.3
```

Comment:
Are you sure that this does not exist? It is part of the stub, though the capitalization here is wrong (<code>Time<del>z</del><ins>Z</ins>one</code>):
```suggestion
    sink(StringUtils.parseTimeZoneString(taint())); // $hasTaintFlow
```
(Additionally its CSV entry is missing)

Reply:
:facepalm: 
I removed this along with the two `parseLocale` functions.